### PR TITLE
Adjust smart content documentation

### PR DESCRIPTION
**Pull request by Prokyonn**
| Q | A
| --- | ---
| Fixed tickets | fixes #issuenum
| Related PRs | sulu/sulu#prnum
| License | MIT

#### What's in this PR?

Explain the contents of the PR.

#### Why?

Which problem does the PR fix?

### CODE REVIEW

File: cookbook/custom-page-data-provider.rst [hunk 1]
```diff
@@ -1,16 +1,15 @@
-Register a custom DataProvider to filter pages by their template
+Custom DataProvider for SmartContent filtering via custom property
```
Comment (nnatter) [inline cookbook/custom-page-data-provider.rst:1]: `Filter pages by a property use a custom SmartContent DataProvider` ?

**Comment by Prokyonn** [comment]
I've changed the custom data provider from `product` to `author` because I think it is more comfortable for readers to just copy the provider and test it out without having to think about a valid property they could filter.

File: cookbook/custom-page-data-provider.rst [hunk 2]
```diff
@@ -7,28 +7,55 @@ comes with a few included providers and also allows to implement new ones to loa
...
+that uses a custom impementation of the ``QueryBuilder``. In the following example a custom ``AuthorPageDataProvider``
+is implemented, which filters the pages for a specific author:
 
 **1. Implement a custom** ``QueryBuilder`` **class that adds conditions**
 
 .. code-block:: php
-
     <?php
-    // src/SmartContent/ProductPageQueryBuilder.php
+    // App\SmartContent\AuthorPageQueryBuilder
```
Comment (nnatter) [inline cookbook/custom-page-data-provider.rst:17]: this is the filename that should be created 🙂 

```suggestion
    // src/SmartContent/AuthorPageQueryBuilder.php
```

File: cookbook/custom-page-data-provider.rst [hunk 3]
```diff
@@ -43,37 +70,38 @@ that uses a custom impementation of the ``QueryBuilder``. To do this, the follow
...
                 - '@sulu_page.smart_content.data_provider.content.proxy_factory'
                 - '@sulu_document_manager.default_session'
                 - '@sulu_page.reference_store.content'
                 - '%sulu_document_manager.show_drafts%'
             tags:
-                - { name: 'sulu.smart_content.data_provider', alias: 'product_pages'}
+                - { name: 'sulu.smart_content.data_provider', alias: 'author_pages' }
 
-**3. Use registered** ``product_pages`` **DataProvider in the template**
+**3. Use registered** ``authr_pages`` **DataProvider in the template**
```
Comment (nnatter) [inline cookbook/custom-page-data-provider.rst:95]:
```suggestion
**3. Use registered** ``author_pages`` **DataProvider in the template**
```

File: cookbook/smart-content-data-provider.rst [hunk 4]
```diff
@@ -221,19 +225,30 @@ avoid filtering for that values.
...
+        public function getConfiguration()
+        {
+            if (!$this->configuration) {
+                $this->configuration = self::createConfigurationBuilder()
+                    ->enableTags()
+                    ->enableLimit()
+                    ->enablePagination()
+                    ->enablePresentAs()
+                    ->enableTypes([
+                        ['type' => 'example-template-1', 'title' => 'my-translation-key-1'],
```
Comment (nnatter) [inline cookbook/smart-content-data-provider.rst:244]:
```suggestion
                        ['type' => 'example-type-1', 'title' => 'my-translation-key-1'],
```

File: cookbook/smart-content-data-provider.rst [hunk 5]
```diff
@@ -221,19 +225,30 @@ avoid filtering for that values.
...
+                        ['type' => 'example-template-2', 'title' => 'my-translation-key-2'],
```
Comment (nnatter) [inline cookbook/smart-content-data-provider.rst:245]:
```suggestion
                        ['type' => 'example-type-2', 'title' => 'my-translation-key-2'],
```

File: cookbook/smart-content-data-provider.rst [hunk 6]
```diff
@@ -279,19 +294,46 @@ avoid filtering for that values.
...
     page.
 
-There is an `enableLimit`, an `enablePagination` and an `enablePresentAs` call,
-which allow you to enable certain features. And the optional `enableView`
-allows you to define to which `View` the application should navigate, when one
-of the resulting items is clicked. The first parameters describes the view
-defined in an `Admin` class, and the second part is a mapping from a json
-pointer, which defines where from the loaded smart content item the value
-should be received, to the parameter of the `View`'s path.
+There are multiple `enable...` calls, which allow you to define certain features.
```
Comment (nnatter) [inline cookbook/smart-content-data-provider.rst:297]:
```suggestion
There are multiple `enable...` calls, which allow you to enable certain features in the administration interface:
```

File: cookbook/custom-page-data-provider.rst (hunk 2 above)
Comment (Prokyonn) [reply cookbook/custom-page-data-provider.rst:17]: Yup I knew that, no idea why I copied the namespace 🙈

**Pull request merged**